packages/core/contracts/financial-templates/optimistic-rewarder/OptimisticRewarder.sol

```diff
@@ -0,0 +1,429 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../../common/implementation/Lockable.sol";
+import "../../common/implementation/Testable.sol";
+import "../../oracle/interfaces/FinderInterface.sol";
+import "../../oracle/interfaces/StoreInterface.sol";
+import "../../oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "../../oracle/implementation/Constants.sol";
+import "../../oracle/interfaces/SkinnyOptimisticOracleInterface.sol";
+import "../../common/implementation/AncillaryData.sol";
+import "../../common/interfaces/AddressWhitelistInterface.sol";
+import "./OptimisticRewarderToken.sol";
+
+/**
+ * @notice The base rewarder contract. This manages depositing rewards and paying them out to tokenholders using values
+ * backed by the OptimisticOracle's dispute process.
+ */
+abstract contract OptimisticRewarderBase is Lockable {
+    using SafeERC20 for IERC20;
+
+    struct RedemptionAmount {
+        uint256 amount;
+        IERC20 token;
+    }
+
+    struct Redemption {
+        uint256 finalFee;
+        uint256 expiryTime;
+    }
+
+    // Constants.
+    FinderInterface public finder;
+    uint256 public liveness;
+    uint256 public bond;
```

Comment:
there seems to be no way to change this bond. is this a problem?

Reply:
So I did it this way to avoid having an owner. We can institute an owner that's assumed to be the deployer if we prefer -- it just introduces a trust assumption on the funds already in the contract. If you can change these parameters, you can effectively pull all the money out.

The way I saw this working was that these parameters would change very infrequently, but when they did, you could migrate this contract to a new contract by passing the NFT into a new OptimisticRewarderNoToken contract and beginning to pay out rewards there.

Thoughts?

Reply:
ye, that makes sence! perhaps add a comment to that effect on why it does not have a modifier.

Reply:
Will add!

Reply:
Done.